cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -76,6 +78,17 @@
   }
 
   /**
+   * If args contains {@link SystemArguments#EMIT_PROGRAM_CONTAINER_METRICS_PREFERENCE}
+   * we evaluate and return the boolean value, if the key is not preset we return false.
+   * @param args runtime arguments
+   * @return true if metrics is enabled
+   */
+  public static boolean isProgramMetricsEnabled(Map<String, String> args) {
+    String metricsEnabled = args.get(EMIT_PROGRAM_CONTAINER_METRICS_PREFERENCE);
+    return metricsEnabled == null ? false : Boolean.valueOf(metricsEnabled);
```

Comment:
Use `return Boolean.parseBoolean(metricsEnabled)` instead, which already handle `null`.

Reply:
Also, see my comment above in the AbstractContext about having a default value for this method.